src/meta/meta_ckpt.c

```diff
@@ -106,25 +106,17 @@ __wt_meta_checkpoint_clear(WT_SESSION_IMPL *session, const char *fname)
 static int
 __ckpt_set(WT_SESSION_IMPL *session, const char *fname, const char *v)
 {
+    WT_DECL_ITEM(tmp);
     WT_DECL_RET;
-    char *config, *newcfg;
-    const char *cfg[3];
 
-    config = newcfg = NULL;
-
-    /* Retrieve the metadata for this file. */
-    WT_ERR(__wt_metadata_search(session, fname, &config));
-
-    /* Replace the checkpoint entry. */
-    cfg[0] = config;
-    cfg[1] = v == NULL ? "checkpoint=()" : v;
-    cfg[2] = NULL;
-    WT_ERR(__wt_config_collapse(session, cfg, &newcfg));
-    WT_ERR(__wt_metadata_update(session, fname, newcfg));
+    WT_ERR(__wt_scr_alloc(session, 0, &tmp));
+    WT_ASSERT(session, strcmp(session->dhandle->name, fname) == 0);
```

Comment:
I don't see the point of this assertion? It's not a requirement they be the same, is it?

Reply:
I am using the metadata string from the dhandle. I wanted to be sure that the metadata I'm using comes from the file we are checkpointing.